### PR TITLE
[improve](cloud) get versions in batch_get

### PR DESCRIPTION
**Pull request by w41ter**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

File: cloud/src/meta-service/meta_service.cpp [hunk 1]
```diff
@@ -326,48 +331,59 @@ void MetaServiceImpl::batch_get_version(::google::protobuf::RpcController* contr
...
+                if (!value.has_value()) {
+                    // return -1 if the target version is not exists.
+                    response->add_versions(-1);
+                } else if (is_table_version) {
+                    if (value->size() != sizeof(int64_t)) {
+                        code = MetaServiceCode::PROTOBUF_PARSE_ERR;
+                        msg = "malformed table version value";
+                        break;
+                    }
+                    int64_t version = *reinterpret_cast<const int64_t*>(value->data());
```
Comment (platoneko) [inline cloud/src/meta-service/meta_service.cpp:376]: If the machine architecture is big endian, byte order needs to be transformed.

**Review by platoneko — CHANGES_REQUESTED** [review]

File: cloud/src/meta-service/meta_service.cpp (hunk 1 above)
Comment (platoneko) [reply cloud/src/meta-service/meta_service.cpp:376]: https://apple.github.io/foundationdb/api-c.html?highlight=fdb_mutation_type_add

Comment (w41ter) [reply cloud/src/meta-service/meta_service.cpp:376]: This PR only focuses on changes related to batch get. For byte order issues, separate PR improvements need to be raised.

**Review by platoneko — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**